app/connector/odata-v2/src/main/java/io/syndesis/connector/odata2/component/ODataComponent.java

```diff
@@ -0,0 +1,311 @@
+
+        Map<String, String> httpHeaders = new HashMap<>();
+        configuration.setHttpHeaders(httpHeaders);
+
+        Methods method = ConnectorOptions.extractOptionAndMap(options, METHOD_NAME, Methods::getValueOf);
+        if (ObjectHelper.isEmpty(method)) {
+            throw new IllegalStateException("No method specified for odata-v2 component");
+        }
+
+        configuration.setMethodName(method.id());
+
+        //
+        // Ensure at least a blank map exists for this property
+        //
+        Map<String, String> endPointHttpHeaders = new HashMap<>();
+        configuration.setEndpointHttpHeaders(endPointHttpHeaders);
+
+        Map<String, Object> resolvedOptions = bundleOptions(options);
+        HttpClientBuilder httpClientBuilder = ODataUtil.createHttpClientBuilder(resolvedOptions);
+        configuration.setHttpClientBuilder(httpClientBuilder);
+
+        HttpAsyncClientBuilder httpAsyncClientBuilder = ODataUtil.createHttpAsyncClientBuilder(resolvedOptions);
+        configuration.setHttpAsyncClientBuilder(httpAsyncClientBuilder);
+
+        if (getServiceUri() != null) {
+            configuration.setServiceUri(getServiceUri());
+        }
+
+        configureResourcePath(configuration, options);
+
+        if (Methods.READ.equals(method)) {
+            //
+            // Modify the query parameters into the expected map
+            //
+            Map<String, String> queryParams = new HashMap<>();
+            if (getQueryParams() != null) {
+                String queryString = getQueryParams();
+                String[] clauses = queryString.split(AMPERSAND, -1);
+                if (clauses.length >= 1) {
+                    for (String clause : clauses) {
+                        String[] parts = clause.split(EQUALS, -1);
+                        if (parts.length == 2) {
+                            queryParams.put(parts[0], parts[1]);
+                        } else if (parts.length < 2) {
+                            queryParams.put(parts[0], EMPTY_STRING);
+                        }
+                        // A clause with more than 1 '=' would be invalid
+                    }
+                }
+            }
+
+            configuration.setQueryParams(queryParams);
+            configuration.setFilterAlreadySeen(isFilterAlreadySeen());
+            configuration.setSplitResult(isSplitResult());
+        }
+
+        Olingo2Component component = new Olingo2Component(getCamelContext());
+        component.setConfiguration(configuration);
+        return Optional.of(component);
+    }
+
+    private void configureResourcePath(Olingo2AppEndpointConfiguration configuration, Map<String, Object> options) {
+        //
+        // keyPredicate is not supported properly in 2.21.0 but is handled
+        // in 2.24.0 by setting it directly on the configuration. Can modify
+        // this when component dependencies are upgraded.
+        //
```

Comment:
This probably ranks as some technical debt that we may or may not ever get around to sorting out?

Reply:
I have tried to use key predicate header directly but soon hit unexpected behavior. So I kept the workaround as it is.